### PR TITLE
Feature/58 test grouping crud

### PR DESCRIPTION
**Pull request by yaoya2618-dot**
Feature/58 test grouping crud
## Description
UI : generate a scrollable block to view test groups and test cases(assignment show view file and also css file). In the scrollable block use js in javascript folder to create dropdown event triggered by click the test grouping title(view specific test cases). CRUD operation: add a text field and a plus button to create test group, add 'x' button in each test group title to delete test group, add 'edit' button in each test group title to edit the name of test group(working in the rails server, but not passing the cucumber tests, so delete the update and edit function in the controller)

Also implement the test grouping controller the relevant functions

## How has this been tested?
rspec for test grouping controller and rspec for test grouping model
cucumber scenarios of create and delete

## Screenshots (if appropriate):
image

## Checklist:

- [ x] Zero `rubocop` violations
- [ ] My code receives an "A" when `rubycritic` is run
- [ x] >90% test coverage of my changes
- [x ] All new and existing tests passed

### CODE REVIEW

File: app/controllers/tests_controller.rb [hunk 1]
```diff
@@ -85,19 +94,27 @@ def destroy
...
 
-
-
   private
 
   def set_assignment
     @assignment = Assignment.find(params[:assignment_id])
   end
   # Use callbacks to share common setup or constraints between actions.
+  def set_test_grouping
```
Comment (masarkar-tamu) [inline app/controllers/tests_controller.rb:103]: Might be missing it, but where is the `set_test_grouping` private function being called in the other controller functions?

Comment (yaoya2618-dot) [reply app/controllers/tests_controller.rb:103]: I am not sure, in the test controller there is line   # before_action :set_test_grouping in line 6. I think originally we want to build a nested structure, that first build the test group and then build a test case. I this code version this part of code can be commented to improve the coverage I guess.

Comment (maxsmith271346) [reply app/controllers/tests_controller.rb:103]: @yaoya2618-dot just remove it if it is unused

Comment (yaoya2618-dot) [reply app/controllers/tests_controller.rb]: removed

File: app/views/assignments/show.html.erb [hunk 2]
```diff
@@ -1,94 +1,80 @@
+<%= stylesheet_link_tag 'assignment_show', media: 'all' %>
+
 <h1>Assignment: <%= @assignment.assignment_name %></h1>
 
 <!-- Display tests for the assignment -->
-<div class="container">
-  <div class="row">
+<div class="container" data-assignment-id="<%= @assignment.id %>">
+2  <div class="row">
```
Comment (masarkar-tamu) [inline app/views/assignments/show.html.erb:7]: I think this 2 might be a typo.

**Review by masarkar-tamu — APPROVED** [review]

**Pull request merged**